src/com/mitrais/dal/TransactionServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package src.com.mitrais.dal;
+
+import src.com.mitrais.model.Account;
+import src.com.mitrais.view.SummaryScreen;
+
+import static src.com.mitrais.utils.Helper.activeAccount;
+import static src.com.mitrais.utils.Helper.destinationAccount;
+import static src.com.mitrais.utils.Helper.transactionAmount;
+import static src.com.mitrais.utils.Helper.generateReferenceNumber;
+
+public class TransactionServiceImpl implements TransactionService {
+    SummaryScreen summaryScreen = new SummaryScreen();
+    @Override
+    public void calculateWithdrawalBalance(Account actAccount, int withdrawAmount) {
+        int balance = actAccount.getBalance() - withdrawAmount;
+        if (withdrawAmount > 1000){
```

Comment:
Why not throw an exception instead? Also, why is the method called `calculateWithdrawalBalance` when it actually does much more? Be consistent with what your code shows as intention

Reply:
I put an exception throwing in this function, but I still don't get what does it mean by "does much more". Could you please define what more refers to here? as my understanding, the function only does set balance and redirect the screen with no extended process. CMIIW

Reply:
If it does set the balance, why not name it properly instead?